OnProfileUpdate/handler.ts

```diff
@@ -0,0 +1,162 @@
+            fiscalCode,
+            previousVersion
+          ),
+          id => dep.profileModel.find([id, fiscalCode])
+        )
+    )
+  );
+
+const deleteProfileEmail = (profileEmail: ProfileEmail) => ({
+  dataTableProfileEmailsRepository
+}: IDependencies): TE.TaskEither<Error, void> =>
+  TE.tryCatch(
+    () => dataTableProfileEmailsRepository.delete(profileEmail),
+    error =>
+      error instanceof Error
+        ? error
+        : new Error("error deleting ProfileEmail from table storage")
+  );
+
+const insertProfileEmail = (profileEmail: ProfileEmail) => ({
+  dataTableProfileEmailsRepository
+}: IDependencies): TE.TaskEither<Error, void> =>
+  TE.tryCatch(
+    () => dataTableProfileEmailsRepository.insert(profileEmail),
+    error =>
+      error instanceof Error
+        ? error
+        : new Error("error inserting ProfileEmail from table storage")
+  );
+
+/*
+If the current email is validated but the previous email was not validated => it inserts the new email into profileEmails
+If the current email is not validated but the previous email was validated => it deletes the previous email from profileEmails
+*/
+const handlePositiveVersion = ({
+  email,
+  fiscalCode,
+  isEmailValidated,
+  version
+}: ProfileDocument): RTE.ReaderTaskEither<
+  IDependencies,
+  Error | CosmosErrors,
+  void
+> =>
+  pipe(
+    getPreviousProfile(fiscalCode, version),
+    RTE.chainW(
+      flow(
+        O.foldW(
+          () => RTE.right<IDependencies, Error, void>(void 0),
+          previousProfile =>
+            isEmailValidated
+              ? previousProfile.isEmailValidated
+                ? RTE.right<IDependencies, Error, void>(void 0)
+                : insertProfileEmail({ email, fiscalCode })
+              : previousProfile.isEmailValidated
+              ? deleteProfileEmail({
+                  email: previousProfile.email,
+                  fiscalCode
+                })
+              : RTE.right<IDependencies, Error, void>(void 0)
+        )
+      )
+    )
+  );
+
+const handleProfile = ({
+  fiscalCode,
+  email,
+  version,
+  isEmailValidated
+}: ProfileDocument): RTE.ReaderTaskEither<
+  IDependencies,
+  Error | CosmosErrors,
+  void
+> =>
+  version === 0
+    ? isEmailValidated
+      ? insertProfileEmail({ email, fiscalCode })
+      : RTE.right<IDependencies, Error, void>(void 0)
+    : handlePositiveVersion({
+        email,
+        fiscalCode,
+        isEmailValidated,
+        version
+      });
+
+export const handler = (documents: ReadonlyArray<unknown>) => async (
+  dependencies: IDependencies
+): Promise<void> => {
+  await pipe(
+    documents.map(document =>
+      pipe(
+        document,
+        ProfileDocument.decode,
+        E.fold(
+          () => TE.right<Error, void>(void 0),
+          profileDocument =>
+            pipe(
+              dependencies,
+              handleProfile(profileDocument),
+              TE.mapLeft(error => {
```

Comment:
In case of a transient error occurrence, it's important not to skip documents. To ensure proper processing and avoid skipping, it's necessary to throw an error rather than bypassing it. 

Additionally, we should verify that both insertion and deletion methods adequately handle retries to maintain robustness.